sentry/src/main/java/io/sentry/SentryOptions.java

```diff
@@ -385,6 +385,9 @@ public class SentryOptions {
 
   private @NotNull IMainThreadChecker mainThreadChecker = NoOpMainThreadChecker.getInstance();
 
+  /** Date provider to retrieve the current date from. */
+  private @NotNull SentryDateProvider dateProvider = new SentryAutoDateProvider();
```

Comment:
M: This likely has to be annotated with internal annotation.

Reply:
Do we not want to allow people to add their own? That said there's a risk of them setting a custom provider and then Sentry static calls don't use the same causing a degradation to `ms` precision. So maybe better to keep it internal, at least for now.